new_client/src/auth/Signin.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { Container, Paper, Grid, TextField, Button } from '@material-ui/core';
+import "./Error.css";
+
+export default function Signin() {
+	const [form, updateForm] = React.useState({
+		username: "",
+		password: "",
+	});
+
+	const [errors, updateErrors] = React.useState({
+		username: "",
+		password: "",
+	});
+
+	function isFormValid() {
+		let formIsValid = true;
+		if (!form.username) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				username: "*Username can't be Empty"
+			}));
+		}
+
+		if (!form.password) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				password: "*Please enter your password."
+			}));
+		}
+		
+		return formIsValid;
+
+	}
+
+
+	function handleChange(event) {
+		const { name, value } = event.target;
+		updateForm(prevDetails => {
+			return (
+				{
+					...prevDetails,
+					[name]: value
+				}
+			);
+		});
+	}
+
+	function handleClick(event) {
+		if (isFormValid()) {
+			event.preventDefault();
+			fetch("http://localhost:8000/api/login", {
```

Comment:
A useful tip: Use a(n environment) variable for all your server URLs, so that you can change it quickly whenever needed.